room_assistance/indico_room_assistance/migrations/20190220_1138_b160812c186b_add_startup_assistance_requests_table.py

```diff
@@ -0,0 +1,90 @@
+"""Add room assistance requests table
+
+Revision ID: b160812c186b
+Revises:
+Create Date: 2019-02-20 11:38:08.691528
+"""
+
+from __future__ import unicode_literals
+
+from alembic import context, op
+from terminaltables import AsciiTable
+
+
+# revision identifiers, used by Alembic.
+revision = 'b160812c186b'
+down_revision = None
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    if context.is_offline_mode():
+        raise Exception('This upgrade is only possible in online mode')
+
+    conn = op.get_bind()
+    conn.execute('''
+        INSERT INTO events.requests (event_id, type, state, data, created_by_id, created_dt)
+        SELECT
+            rvl.linked_event_id,
+            'room-assistance',
+            1,
+            '{}',
+            (SELECT id FROM users.users WHERE is_system),
+            rv.start_dt
+        FROM roombooking.reservations rv
+        JOIN roombooking.reservation_links rvl ON rv.link_id = rvl.id AND rvl.link_type = 2
+        WHERE rv.needs_assistance
+    ''')
+
+    invalid_reservations = conn.execute('''
+        SELECT rv.id, rv.room_id, rvl.id, rv.start_dt, rv.end_dt, rv.booked_for_name
+        FROM roombooking.reservations rv
+        LEFT JOIN roombooking.reservation_links rvl ON rv.link_id = rvl.id
+        WHERE rv.needs_assistance
+        AND (rv.link_id IS NULL OR rvl.link_type != 2)
+        AND rv.start_dt > NOW()
+    ''')
+
+    table_data = [['ID', 'Room ID', 'Link ID', 'Start date', 'End date', 'Booked for']]
+    for resv in invalid_reservations:
+        table_data.append(list(resv))
+
+    if len(table_data) > 1:
+        print(AsciiTable(table_data,
+                         'Reservations without a link to an event that need assistance in the future').table)
+
+    conn.execute('''
+        UPDATE settings SET module = 'plugin_room_assistance', name = 'room_assistance_recipients'
```

Comment:
did you test this? the table names here are missing the schema so the queries fail, so if this code actually works for you something is screwed up with your postgres search_path

Reply:
isn't the `indico` schema the default one?

Reply:
Now I remember, the search path is `"$user", public` by default. So if your user is called indico it works for you. That's why your DB user in development should not be called indico ;)

Reply:
Oh. I should have checked that..